### PR TITLE
minio: Change config properties and defaults

### PR DESCRIPTION
**Pull request by krisis**
Modifications
    - service-port --> port (9000)
    - data-size      --> volume-size (1024 MiB)
    - service.cpus --> resource.cpus (1)
    - service.mem --> resource.mem (1024 MiB)
    
    Removal
    - storage.config-size
    - networking.https-redirect
    
    Additions
    - networking.public-access (true)

### CODE REVIEW

**Comment by joerg84** [comment]
LGTM

File: repo/packages/M/minio/6/config.json [hunk 1]
```diff
@@ -0,0 +1,80 @@
+{
+    "$schema": "http://json-schema.org/schema#",
+    "properties": {
+        "service": {
+            "type": "object",
+            "description": "Service properties for Minio on DC/OS",
+            "properties": {
+                "name": {
+                    "description": "Name of this service instance",
```
Comment (ryadav88) [inline repo/packages/M/minio/6/config.json]: add a default for the service name

File: repo/packages/M/minio/6/config.json [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+                "mem"
+            ]
+        },
+        "storage": {
+            "type": "object",
+            "description": "Storage properties for Minio on DC/OS",
+            "properties": {
+                "data-size": {
+                    "type": "number",
+                    "description": "Size of data volume in MiB"
```
Comment (ryadav88) [inline repo/packages/M/minio/6/config.json]: add default value

File: repo/packages/M/minio/6/config.json [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+                }
+            }
+        },
+        "networking": {
+            "description": "Networking properties for Minio on DC/OS",
+            "type": "object",
+            "properties": {
+                "service-port": {
+                    "description": "Minio service port",
+                    "type": "number"
```
Comment (ryadav88) [inline repo/packages/M/minio/6/config.json]: add default.

File: repo/packages/M/minio/6/config.json (hunk 2 above)
Comment (krisis) [inline repo/packages/M/minio/6/config.json]: `data-size` and `service-port` have no defaults only to force the user to input specific values before installation.

File: repo/packages/M/minio/6/config.json (hunk 1 above)
Comment (krisis) [inline repo/packages/M/minio/6/config.json]: Service name is without defaults to force the user to give a unique name for each instance and avoid having many instances with the default service name.

File: repo/packages/M/minio/6/config.json (hunk 2 above)
Comment (ryadav88) [inline repo/packages/M/minio/6/config.json]: @krisis defaults values are needed to the the `default installation` to work. You can add recommendations in the `description` sections for all these properties.

Comment (krisis) [inline repo/packages/M/minio/6/config.json]: @ryadav88 OK, I shall add default values where missing.

File: repo/packages/M/minio/6/config.json (hunk 3 above)
Comment (krisis) [inline repo/packages/M/minio/6/config.json]: @ryadav88 `service-port` configuration option having a default is problematic when 2 or more instances of minio are installed with the same (default) port. IIUC Marathon/Marathon-LB don't reject instances using the same service port. Now, we are left to choose between one-click default installation and incorrect service deployment. I am inclined towards preventing incorrect service deployment by _not_ providing a default for `service-port`. If you think it is uncommon for users to make this mistake, then I shall add a default. Let me know your thoughts on this.

**Comment by krisis** [comment]
@ryadav88 With the recent changes, all configuration options have default values, allowing for a single-click default installation.

File: repo/packages/M/minio/6/package.json [hunk 4]
```diff
@@ -0,0 +1,26 @@
+{
+  "packagingVersion": "3.0",
+  "name": "Minio",
```
Comment (ryadav88) [inline repo/packages/M/minio/6/package.json]: change `Minio` -> `minio`
Changing the name here creates a different package in Universe.

File: repo/packages/M/minio/6/package.json [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+  "maintainer": "dev@minio.io",
+  "website": "https://www.minio.io",
+  "framework": false,
+  "description": "Minio is an object storage server for cloud native applications. To deploy at scale, use DC/OS to orchestrate Minio containers one per tenant.",
+  "tags": [
+    "data",
+    "storage",
+    "filesystem"
+  ],
+  "preInstallNotes": "There is no limit to number of Minio services you can launch.",
```
Comment (ryadav88) [inline repo/packages/M/minio/6/package.json]: Add this in front of `preInstallNotes`: `This DC/OS Service is currently in preview. There may be bugs, incomplete features, incorrect documentation, or other discrepancies. Preview packages should never be used in production!`

Comment (ryadav88) [inline repo/packages/M/minio/6/package.json]: Please update the minio example in `dcos/examples` with the changes and how to use the package.

Comment (ryadav88) [inline repo/packages/M/minio/6/package.json]: Add a link to the example at the end of `preInstallNotes`

Comment (krisis) [inline repo/packages/M/minio/6/package.json]: @ryadav, we wish to remove the warning about this package being preview and not production ready. Minio by itself is production ready, is there any criteria for a universe package to meet before calling itself production ready?

I will add example and a link to it once the example PR merges. Does that make sense?

**Comment by krisis** [comment]
@ryadav88 Please find the PR on examples repo corresponding to this change here - https://github.com/dcos/examples/pull/171

**Comment by joerg84** [comment]
@ryadav88 PTAL

File: repo/packages/M/minio/6/package.json (hunk 5 above)
Comment (ryadav88) [inline repo/packages/M/minio/6/package.json]: @krisis this message is for the integration and not for the base tech and is required for all packages that are not in beta or selected. We can discuss offline on how to make it beta ready. Please add a link to the example and I can merge both the PRs at the same time. 
Other changes look good. Will merge the PR when these comments are addressed.

Comment (krisis) [inline repo/packages/M/minio/6/package.json]: @ryadav88 I have added the warning at the front and link to example at the end of `preInstallNotes`.

**Comment by krisis** [comment]
@ryadav88 I have made changes to the preInstallNotes as recommended. It is ready for review.

**Pull request merged**